SwiftSnake/SwiftSnake/Extensions/GameLogicControllers/CollisionHandler.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  CollisionHandler.swift
+//  SwiftSnake
+//
+//  Created by Ellen Shapiro (Vokal) on 9/24/15.
+//  Copyright © 2015 Vokal. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+struct CollisionHandler {
+    
+    //MARK: - Convenience wrappers around category bitmask comparison
+    
+    private static func isFood(body: SKPhysicsBody) -> Bool {
+        return body.categoryBitMask == NodeBitmask.food
+    }
+    
+    private static func isSnakeHead(body: SKPhysicsBody) -> Bool {
+        return body.categoryBitMask == NodeBitmask.snakeHead
+    }
+    
+    private static func isSnakeBody(body: SKPhysicsBody) -> Bool {
+        return body.categoryBitMask == NodeBitmask.snakeBody
+    }
+    
+    //MARK: - Major collision detectors
+    
+    /**
+    - parameter firstBody: The first body to compare
+    - parameter secondBody: The second body to compare the first body to
+    - returns: true if the snake head has run into the snake tail. false if not.
+    */
+    static func isOuroboros(firstBody: SKPhysicsBody, _ secondBody: SKPhysicsBody) -> Bool {
+        if isSnakeHead(firstBody) {
```

Comment:
a more "swifty" way to do this could be to use a switch statement on both of the body bit masks:
`switch (firstBody.categoryBitmask, secondBody.categoryBitmask) {...`